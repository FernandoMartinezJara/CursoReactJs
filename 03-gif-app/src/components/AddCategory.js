import React, { useState} from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({setcategories}) => {

    const [inputValue, setinputValue] = useState('')

    const handleInputChange = (e) =>{
        setinputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim()){
            setcategories(cats => [inputValue, ...cats]);
            setinputValue('')
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{ inputValue }</p>
            <input
                value={inputValue}
                onChange={ handleInputChange }
                type="text"/>
        </form>
    )
}

AddCategory.propTypes = {
    setcategories: PropTypes.func.isRequired
}
