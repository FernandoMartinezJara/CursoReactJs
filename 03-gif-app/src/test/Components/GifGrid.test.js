import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas a componente <GifGrid/>', () => {

    const category = 'Bon Jovi';
    
    test('debe de mostrar correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot();
        
    })


    test('debe de mostrar items cuando se cargan imagenes con useFetchGifs', () => {
        
        const imgs = [{
            id: 'ABC',
            url: 'https://www.google.cl',
            title: 'Cantantes'
        }];

        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false

        })
        const wrapper = shallow(<GifGrid category={category}/>)

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(imgs.length);
    })
    
    
})
