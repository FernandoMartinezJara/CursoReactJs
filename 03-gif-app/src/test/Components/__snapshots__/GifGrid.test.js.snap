// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a componente <GifGrid/> debe de mostrar correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    Bon Jovi
  </h3>
  <p
    className="card animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
